bluepysnap/nodes/node_population.py

```diff
@@ -269,14 +269,16 @@ def _get_node_set(self, node_set_name):
         """Returns the node set named 'node_set_name'."""
         if node_set_name not in self._node_sets:
             raise BluepySnapError(f"Undefined node set: '{node_set_name}'")
-        return self._node_sets[node_set_name]
+        return self._node_sets.resolve_ids(node_set_name, self._population)
 
     def _resolve_nodesets(self, queries):
         def _resolve(queries, queries_key):
             if queries_key == query.NODE_SET_KEY:
                 if query.AND_KEY not in queries:
                     queries[query.AND_KEY] = []
-                queries[query.AND_KEY].append(self._get_node_set(queries[queries_key]))
+                queries[query.AND_KEY].append(
+                    {query.NODE_ID_KEY: self._get_node_set(queries[queries_key])}
+                )
```

Comment:
Very good that the times are better, but I'm surprised that querying on Layer23Inhibitory is taking 1 second.

I did a quick check, and I see that most of the time is spent in doing 9 times (one for each mtype) something like:
```python
%time np.in1d(cells['mtype'], 'L23_SBC')
```
that takes around 110ms:
```
CPU times: user 94.2 ms, sys: 17.5 ms, total: 112 ms
Wall time: 111 ms
array([False, False, False, ..., False, False, False])
```
and it's not optimal because:
- mtype is a category, but using np.in1d requires a conversion to object (35ms), and then looking through strings instead of numbers (65ms)
- it should be called only once, passing all the mtypes as a list

For comparison, this calls 
```python
 %time cells['mtype'].isin(['L23_SBC', 'L23_BTC', 'L23_NGC','L23_MC','L23_NBC','L23_LBC','L23_BP','L23_CHC', 'L23_DBC']).to_numpy()
```
would require only 9ms to check all the 9 mtypes at the same time:
```
CPU times: user 9.02 ms, sys: 3 µs, total: 9.02 ms
Wall time: 8.44 ms
array([False, False, False, ..., False, False, False])
```

However, these considerations are unrelated to the MR. I can see if the np.in1d can be safely changed to make the query function more performant, and maybe we can repeat the comparison.

Reply:
Thanks for profiling. 

> I did a quick check, and I see that most of the time is spent in doing 9 times (one for each mtype) something like:
> ```python
> %time np.in1d(cells['mtype'], 'L23_SBC')
> ```

I am not surprised. I guess there has not been a reason to optimize the queries. And, to be fair, 1s for a query is still okayish provided how much data it filters. Obviously, if we can easily reduce it to tenth of that, we should.

> I can see if the np.in1d can be safely changed to make the query function more performant, and maybe we can repeat the comparison.

Am I assuming correctly that you would rather keep the node set resolution in SNAP?

Reply:
> Am I assuming correctly that you would rather keep the node set resolution in SNAP?

Nope, but it can be worth to improve the query function because it's used also when `get()` is called with generic queries

Reply:
Ah, of course! Makes sense.

Reply:
Do you mind repeating the same benchmark with version 1.0.7, after merging or rebasing on master?

Reply:
with `snap`:
```
# ROUNDS: 100

# Layer23Inhibitory #
- total time: 5.96 s
- per round (avg): 0.060s

# S1HL #
- total time: 2.55 s
- per round (avg): 0.025s

# mc2;VPL #
- total time: 0.66 s
- per round (avg): 0.007s

# mc2_Column #
- total time: 6.51 s
- per round (avg): 0.065s

--------------------------------
# ROUNDS: 1000

# Layer23Inhibitory #
- total time: 27.48 s
- per round (avg): 0.027s

# S1HL #
- total time: 19.42 s
- per round (avg): 0.019s

# mc2;VPL #
- total time: 6.56 s
- per round (avg): 0.007s

# mc2_Column #
- total time: 66.19 s
- per round (avg): 0.066s
```

with `libsonata`:
```
# ROUNDS: 100

# Layer23Inhibitory #
- total time: 3.39 s
- per round (avg): 0.034s

# S1HL #
- total time: 9.93 s
- per round (avg): 0.099s

# mc2;VPL #
- total time: 3.95 s
- per round (avg): 0.040s

# mc2_Column #
- total time: 7.02 s
- per round (avg): 0.070s

--------------------------------
# ROUNDS: 1000

# Layer23Inhibitory #
- total time: 26.13 s
- per round (avg): 0.026s

# S1HL #
- total time: 103.41 s
- per round (avg): 0.103s

# mc2;VPL #
- total time: 40.42 s
- per round (avg): 0.040s

# mc2_Column #
- total time: 71.51 s
- per round (avg): 0.072s
```